src/views/design/ComponentDetail/index.tsx

```diff
@@ -598,15 +597,22 @@ const ComponentDetail = withStyles(styles)(
                     mandatory={mandatory}
                     isCreateParameter={isAddingParameter}
                     onEdit={(newParameter) => {
-                        const newParameters = [...newComponentDetail.parameters];
                         if (!isAddingParameter) {
+                            const newParameters = [...newComponentDetail.parameters];
                             newParameters[editParameterIndex] = newParameter;
+                            const orderedParameters = orderComponentParameter(newParameters, matchingComponentType);
+                            this.updateComponent({
+                                parameters: orderedParameters,
+                            });
+                        }
+                        if (isAddingParameter) {
+                            const newParameters = [...newComponentDetail.parameters, newParameter];
+                            newParameters[editParameterIndex] = newParameter;
+                            const orderedParameters = orderComponentParameter(newParameters, matchingComponentType);
+                            this.updateComponent({
+                                parameters: orderedParameters,
+                            });
                         }
```

Comment:
you have duplicate logic in both the if and else clause. You can move that outside of the if-else branch